data/dataset.py

```diff
@@ -144,10 +174,18 @@ def load_cache_dataset(
 
     print("Building translation dataset from", dataset_name, split)
     midi_dataset = load_dataset(dataset_name, split=split)
-    translation_dataset = build_translation_dataset(
+
+    # hardcoded maximum shard size as 5000
+    num_shards = len(midi_dataset) // 5000 + 1
+    translation_dataset = shard_and_build(
         dataset=midi_dataset,
+        dataset_name=dataset_name,
         dataset_cfg=dataset_cfg,
+        num_shards=num_shards,
+        dataset_cache_path=dataset_cache_path,
     )
-    translation_dataset.save_to_disk(dataset_cache_path)
+
```

Comment:
This has  too many arguments. You don't need to pass `dataset_name` (if you need to attach this information after building, you can do it outside of `shard_and_build`) and if you're hardocing sharding parameters, it's better to hide it inside the sharding function.

Reply:
We need to add dataset_name to dataset.info before build_translation_dataset, because we append dataset.info.dataset_name to each sequence.

Reply:
Are you sure you need to add it though? I tried locally, and looks like shards are inheriting the parent dataset info. But if that's not the case, I think you still don't need to do this inside `shard_and_build`, but rather after it:

```python
translation_dataset = shard_and_build(...)
translation_dataset.info.dataset_name = dataset_name
```

That way you're avoid unnecessary obfuscation

Reply:
I had some problems with sharding when I added it before, I will try again, thanks!

Reply:
Adding dataset_name to dataset.info before sharding:
```
midi_dataset.info.dataset_name = dataset_name
    # hardcoded maximum shard size as 5000
    num_shards = len(midi_dataset) // 5000 + 1
    translation_dataset = shard_and_build(
        dataset=midi_dataset,
        dataset_name=dataset_name,
        dataset_cfg=dataset_cfg,
        num_shards=num_shards,
        dataset_cache_path=dataset_cache_path,
    )
```
produces an error when calling dataset.shard:
TypeError: DatasetInfo.__init__() got an unexpected keyword argument 'dataset_name'

and btw I am lost as to why did it work before: maestro-v1-sustain also does not seem to have this attribute inside it's "info".